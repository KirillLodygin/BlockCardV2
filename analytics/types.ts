import { NOTIFICATION_TYPE } from '@vtb/ui-kit/adaptive/molecules/notification';

import { LOCK_OPERATION } from '../types';
import {
  CREDIT_ACTION_PERMANENT_BLOCK_STRING,
  CREDIT_ACTION_TEMPORARY_BLOCK_STRING,
  CREDIT_CARD_ACTION_MAP,
  CREDIT_PRODUCT,
  CREDIT_STREAM,
  DEBIT_ACTION_PERMANENT_BLOCK_STRING,
  DEBIT_ACTION_TEMPORARY_BLOCK_STRING,
  DEBIT_CARD_ACTION_MAP,
  DEBIT_PRODUCT,
  DEBIT_STREAM,
} from './constants';

export type AnalyticAction = {
  text: string;
  target: string;
};

// для аналитики
export enum CARD_ACTION_TYPE {
  UNFREEZE = 'unfreeze',
  BLOCK = 'block',
}

export type AnalyticsMapType = {
  [x in string]: {
    [x in string]: {
      text: string;
      target: string;
    };
  };
};

export type AnalyticsEntityType = {
  stream: typeof CREDIT_STREAM | typeof DEBIT_STREAM;
  product: typeof CREDIT_PRODUCT | typeof DEBIT_PRODUCT;
  getActionEntityString: (
    blockType: LOCK_OPERATION,
  ) =>
    | typeof DEBIT_ACTION_TEMPORARY_BLOCK_STRING
    | typeof DEBIT_ACTION_PERMANENT_BLOCK_STRING
    | typeof CREDIT_ACTION_TEMPORARY_BLOCK_STRING
    | typeof CREDIT_ACTION_PERMANENT_BLOCK_STRING;
  analyticsMap: AnalyticsMapType;
  type: typeof CARD_ACTION_TYPE.BLOCK | typeof CARD_ACTION_TYPE.UNFREEZE;
};

enum BLOCK_DEBIT_CARD_SCREEN {
  CHECK = 'CHECK',
  RESULT = 'RESULT',
  START = 'START',
}

export type NotificationType = typeof NOTIFICATION_TYPE.info | typeof NOTIFICATION_TYPE.fail;

export type AnalyticsOptionsType = {
  blockType: LOCK_OPERATION;
  cardName: string;
  screenData: BLOCK_DEBIT_CARD_SCREEN;
  notificationType?: NotificationType;
  reason?: string;
};

export type DebitAnalyticsMapType = typeof DEBIT_CARD_ACTION_MAP;
export type CreditAnalyticsMapType = typeof CREDIT_CARD_ACTION_MAP;

export type NotificationKeys = keyof typeof DEBIT_CARD_ACTION_MAP;
