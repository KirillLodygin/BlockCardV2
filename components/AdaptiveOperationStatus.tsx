import React from 'react';
import styled from 'styled-components/macro';

import { AdditionalTypography, ADDITIONAL_TYPOGRAPHY_SIZE } from '@vtb/ui-kit/adaptive/atoms/additionalTypography';
import { MainTypography, MAIN_TYPOGRAPHY_SIZE } from '@vtb/ui-kit/adaptive/atoms/mainTypography';
import { NOTIFICATION_TYPE } from '@vtb/ui-kit/adaptive/molecules/notification';
import { INDENT } from '@vtb/ui-kit/tokens/sizes';
import { TYPOGRAPHY_TAG } from '@vtb/ui-kit/tokens/typography';

import {
  OperationFormStatusAdaptive,
  StatusIconWrapperAdaptive,
  StatusInfoWrapper,
} from '@common/overmind/layouts/common/components/operationForm/AdaptiveOperationFormStatus';
import { ErrorAnimation, SuccessAnimation } from '@common/overmind/layouts/common/layoutParts/NotificationAnimation';

type NotificationType = typeof NOTIFICATION_TYPE.info | typeof NOTIFICATION_TYPE.fail;

type Props = {
  description: string;
  title: string;
  type: NotificationType;
};

const OperationFormStatusAdaptiveWrapper = styled(OperationFormStatusAdaptive)`
  margin-bottom: ${INDENT.m};
`;

const StatusTextAdaptive = styled(MainTypography)`
  margin-bottom: ${INDENT.xxxs};
`;

export const AdaptiveOperationStatus = ({ description, title, type }: Props) => (
  <OperationFormStatusAdaptiveWrapper isError={type === NOTIFICATION_TYPE.fail}>
    <StatusIconWrapperAdaptive>
      {type === NOTIFICATION_TYPE.fail ? <ErrorAnimation /> : <SuccessAnimation />}
    </StatusIconWrapperAdaptive>
    <StatusInfoWrapper>
      <StatusTextAdaptive tag={TYPOGRAPHY_TAG.h2} size={MAIN_TYPOGRAPHY_SIZE.m} autoFocus>
        {title}
      </StatusTextAdaptive>
      <AdditionalTypography size={ADDITIONAL_TYPOGRAPHY_SIZE.m}>{description}</AdditionalTypography>
    </StatusInfoWrapper>
  </OperationFormStatusAdaptiveWrapper>
);
