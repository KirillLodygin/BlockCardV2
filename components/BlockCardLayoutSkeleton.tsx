import React from 'react';
import { Flex, Box } from 'reflexbox/styled-components';

import { AdaptiveCol, AdaptiveRow } from '@vtb/ui-kit/adaptive/atoms/layout';
import { FIELD_WIDTH, INDENT } from '@vtb/ui-kit/tokens/sizes';

import {
  FieldSkeleton,
  DropdownSkeleton,
  TextlineSkeleton,
  LINE_SIZES,
  SKELETON_LINE_WIDTH,
} from '@components/Skeleton/components';
import { ButtonSkeleton } from '@components/Skeleton/components/adaptive/ButtonSkeleton';

const BlockCardLayoutSkeleton = () => (
  <Box mt={INDENT.m}>
    <AdaptiveRow>
      <AdaptiveCol tablet={8}>
        <Flex flexDirection="column">
          <TextlineSkeleton width={SKELETON_LINE_WIDTH.xxl} lineSize={LINE_SIZES.l} />
          <Box mt={INDENT.sm}>
            <FieldSkeleton width={FIELD_WIDTH.auto} />
          </Box>
          <Box mt={INDENT.xxs}>
            <DropdownSkeleton width={FIELD_WIDTH.auto} />
          </Box>
          <Box mt={INDENT.l}>
            <FieldSkeleton width={FIELD_WIDTH.auto} />
          </Box>
          <Box mt={INDENT.xxs}>
            <ButtonSkeleton />
          </Box>
          <Box mt={INDENT.l}>
            <TextlineSkeleton width={SKELETON_LINE_WIDTH.xs} lineSize={LINE_SIZES.l} />
          </Box>
        </Flex>
      </AdaptiveCol>
    </AdaptiveRow>
  </Box>
);

export { BlockCardLayoutSkeleton };
