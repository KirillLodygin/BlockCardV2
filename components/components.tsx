import React, { useEffect, useRef, useState } from 'react';
import { Box, Flex } from 'reflexbox/styled-components';
import styled from 'styled-components/macro';

import { AdditionalTypography, ADDITIONAL_TYPOGRAPHY_SIZE } from '@vtb/ui-kit/adaptive/atoms/additionalTypography';
import { ButtonText } from '@vtb/ui-kit/adaptive/atoms/button';
import { Icon, ICON_SIZE } from '@vtb/ui-kit/adaptive/atoms/icon';
import { MAIN_TYPOGRAPHY_SIZE, MainTypography } from '@vtb/ui-kit/adaptive/atoms/mainTypography';
import { Notification as Notice, NOTIFICATION_TYPE } from '@vtb/ui-kit/adaptive/molecules/notification';
import { COLOR_GETTERS } from '@vtb/ui-kit/adaptive/tokens/palette/colorGetters';
import { CardPending, ChevronLeft } from '@vtb/ui-kit/assets';
import { INDENT } from '@vtb/ui-kit/tokens/sizes';
import { TYPOGRAPHY_WEIGHT } from '@vtb/ui-kit/tokens/typography';

import { ReactComponent as CardBlockErrorDark } from '@assets/cards/cardBlockErrorDark.svg';
import { ReactComponent as CardBlockErrorLight } from '@assets/cards/cardBlockErrorLight.svg';
import { ReactComponent as CardBlockSuccessDark } from '@assets/cards/cardBlockSuccessDark.svg';
import { ReactComponent as CardBlockSuccessLight } from '@assets/cards/cardBlockSuccessLight.svg';
import { useThemedContent } from '@common/appTheme';
import { OPERATIONS_CONSTANTS } from '@common/overmind/constants/base';
import { DEFAULT_ERROR_MSG } from '@common/overmind/constants/messages';
import { OperationFormChild } from '@common/overmind/layouts/common/components/operationForm';
import {
  OperationFormStatus,
  StatusIconContainer,
  StatusText,
} from '@common/overmind/layouts/common/components/operationForm/OperationFormStatus';
import { ELEMENT_TYPES } from '@common/overmind/layouts/common/constants';
import { cutBBCodesAndTags, parseBBCode } from '@common/parse-bb-code';
import { useSetNavBarTitle } from '@modules/NavBar';

// TODO-gordeev: использовать более подходящие типы после того как они будут сформированы в овермайнде
// TODO-svetlov: перетащить все компоненты в свои файлы и переместить в папку с компонентами

type TitleProps = {
  titleText?: string;
};

export const Title = ({ titleText }: TitleProps) => (
  <Box mb={INDENT.sm}>
    <MainTypography autoFocus>{parseBBCode(titleText)}</MainTypography>
  </Box>
);

const NotificationText = styled(AdditionalTypography).attrs({
  size: ADDITIONAL_TYPOGRAPHY_SIZE.m,
})`
  color: ${COLOR_GETTERS.text.secondary};
`;

type NotificationProps = {
  hasErrorNotification: boolean;
  isLockingSuccess: boolean;
  statusText: string | undefined;
  details: React.Component;
};

export const Notification = ({ statusText, hasErrorNotification, isLockingSuccess, details }: NotificationProps) => {
  const illustrations = useThemedContent(
    {
      success: <CardBlockSuccessLight />,
      error: <CardBlockErrorLight />,
    },
    {
      success: <CardBlockSuccessDark />,
      error: <CardBlockErrorDark />,
    },
  );

  return (
    <Box mb={INDENT.l}>
      <OperationFormStatus>
        <StatusIconContainer>
          {isLockingSuccess && illustrations.success}
          {hasErrorNotification && illustrations.error}
        </StatusIconContainer>
        <StatusText autoFocus>{statusText}</StatusText>

        <NotificationText>{details}</NotificationText>
      </OperationFormStatus>
    </Box>
  );
};

type ButtonsBlockProps = {
  buttons: ReadonlyArray<any>;
};

export const ButtonsBlock = ({ buttons }: ButtonsBlockProps) => (
  <Box mt={INDENT.m}>
    {buttons.map((button) =>
      button.props.name === OPERATIONS_CONSTANTS.commandNames.CANCEL ? (
        <Box mt={INDENT.l} key={button.key}>
          {button}
        </Box>
      ) : (
        <Box key={button.key}>{button}</Box>
      ),
    )}
  </Box>
);

type ElementsBlockProps = {
  elements: ReadonlyArray<any>;
};

export const ElementsBlock = ({ elements }: ElementsBlockProps) => (
  <Box mb={INDENT.xs} flexDirection="column" key="rootGroup">
    {elements.map((field, i) => (
      <OperationFormChild type={ELEMENT_TYPES.FIELD} key={i}>
        {field}
      </OperationFormChild>
    ))}
  </Box>
);

export const BlockTitle = ({ titleText }: TitleProps) => {
  const TitleRef = useRef(null);
  useSetNavBarTitle(titleText, TitleRef);

  if (titleText) {
    return (
      <Box mb={INDENT.m}>
        <MainTypography size={MAIN_TYPOGRAPHY_SIZE.m} tag="h3" ref={TitleRef} autoFocus>
          {titleText}
        </MainTypography>
      </Box>
    );
  }

  return null;
};

export const DefaultErrorBlock = () => (
  <Box mt={INDENT.m}>
    <Notice text={DEFAULT_ERROR_MSG} type={NOTIFICATION_TYPE.fail} />
  </Box>
);

const StyledAdditionalTypography = styled(AdditionalTypography)`
  color: ${COLOR_GETTERS.text.primary};
`;

type CodeRequestTimerProps = {
  isTimerRunning: boolean;
  setIsTimerRunning: (value: boolean) => void;
  text: string;
  timeout: number;
};

export const CodeRequestTimer = ({ isTimerRunning, setIsTimerRunning, text, timeout }: CodeRequestTimerProps) => {
  const [seconds, setSeconds] = useState(timeout);

  useEffect(() => {
    const timerId = setInterval(() => {
      setSeconds(seconds - 1);
    }, 1000);

    if (seconds === 0) {
      clearInterval(timerId);
      setIsTimerRunning(false);
    }

    return () => clearInterval(timerId);
  }, [seconds, setIsTimerRunning]);

  useEffect(() => {
    if (isTimerRunning) {
      setSeconds(timeout);
    }
  }, [isTimerRunning, timeout]);

  return (
    <Box mt={INDENT.xs}>
      <Flex>
        {isTimerRunning && <Icon type={CardPending} size={ICON_SIZE.s} />}
        <StyledAdditionalTypography weight={TYPOGRAPHY_WEIGHT.bold} role="timer">
          <Box as="span" ml={INDENT.xxxxs}>
            {cutBBCodesAndTags(text, seconds.toString().padStart(2, '0'))}
          </Box>
        </StyledAdditionalTypography>
      </Flex>
    </Box>
  );
};

type BackButtonBlockProps = {
  children: string;
  'data-test-id': string;
  disabled?: boolean;
  onClick: () => void;
};

export const BackButtonBlock = ({ children, ...props }: BackButtonBlockProps) => (
  <ButtonText size="medium" icon={ChevronLeft} {...props}>
    {children}
  </ButtonText>
);
