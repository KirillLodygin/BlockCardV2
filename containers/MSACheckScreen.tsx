import React, { useCallback, useState } from 'react';
import { Box } from 'reflexbox/styled-components';

import { getDataTestId } from '@vtb/ui-kit/_utils/getDataTestId';
import { AdditionalTypography } from '@vtb/ui-kit/adaptive/atoms/additionalTypography';
import { ButtonContained } from '@vtb/ui-kit/adaptive/atoms/button';
import { ControlWithTips } from '@vtb/ui-kit/adaptive/atoms/controlWithTips';
import { DataRowTable, DATA_ROW_SIZES } from '@vtb/ui-kit/adaptive/atoms/dataRow';
import { Input } from '@vtb/ui-kit/adaptive/atoms/input';
import { MAIN_TYPOGRAPHY_SIZE } from '@vtb/ui-kit/adaptive/atoms/mainTypography';
import { FIELD_WIDTH, INDENT } from '@vtb/ui-kit/tokens/sizes';

import { CheckScreenInputSkeleton } from '../components/CheckScreenInputSkeleton';
import { BackButtonBlock, BlockTitle, CodeRequestTimer } from '../components/components';
import { BLOCK_CARD_SMS_CODE_LENGTH, SCENARIO_TEST_ID } from '../constants/commonConstants';
import { BUTTONS_LOCK, LOCK_OPERATION, RowTableData } from '../types';

type Props = {
  content: {
    backButton: {
      title: string;
      type: BUTTONS_LOCK;
    };
    continueButton: {
      title: string;
      type: BUTTONS_LOCK;
    };
    dataRowTableData: RowTableData[];
    inputErrorText: string;
    inputLabel: string;
    phoneInfo: string;
    resendButton: {
      title: string;
      type: BUTTONS_LOCK;
    };
    timerInfo: string;
    title: string;
  };
  blockCardType: LOCK_OPERATION;
  isFetchLoading: boolean;
  isFetchAthorizeLoading: boolean;
  onBack: () => void;
  onChangeSecureCode: (value: string) => void;
  onSendSmsCode: (isSentCode: boolean) => void;
  secureCode: string;
  onResetSecureCodeError: (isError: boolean) => void;
  isSecureCodeInputError: boolean;
  timeout: number;
};

export const MSACheckScreen = ({
  content: {
    backButton,
    continueButton,
    dataRowTableData,
    inputErrorText,
    inputLabel,
    phoneInfo,
    resendButton,
    timerInfo,
    title,
  },
  blockCardType,
  isFetchLoading,
  isFetchAthorizeLoading,
  onBack,
  onChangeSecureCode,
  onSendSmsCode,
  secureCode,
  onResetSecureCodeError,
  isSecureCodeInputError,
  timeout,
}: Props) => {
  const [isTimerRunning, setIsTimerRunning] = useState(true);
  const [isSentCode, setIsSentCode] = useState(blockCardType === LOCK_OPERATION.UNFREEZE);

  const handleSendSmsCode = useCallback(() => {
    onSendSmsCode(isSentCode);
    setIsSentCode(true);
  }, [isSentCode, onSendSmsCode]);

  const handleResentSmsCode = useCallback(() => {
    onSendSmsCode(isSentCode);
    setIsTimerRunning(true);
    onResetSecureCodeError(false);
  }, [isSentCode, onResetSecureCodeError, onSendSmsCode]);

  return (
    <>
      <BlockTitle titleText={title} />
      <Box mb={INDENT.m}>
        <DataRowTable data={dataRowTableData} size={DATA_ROW_SIZES.m} />
      </Box>
      {isSentCode &&
        (isFetchAthorizeLoading ? (
          <CheckScreenInputSkeleton />
        ) : (
          <Box mb={INDENT.l}>
            <Box mb={INDENT.sm}>
              <AdditionalTypography size={MAIN_TYPOGRAPHY_SIZE.m} tag="p" weight="normal">
                {phoneInfo}
              </AdditionalTypography>
            </Box>
            <Box maxWidth={FIELD_WIDTH.xl}>
              <ControlWithTips width={FIELD_WIDTH.m} errorText={inputErrorText} hasError={isSecureCodeInputError}>
                <Input
                  data-test-id={getDataTestId(SCENARIO_TEST_ID, 'input_smscode_field')}
                  disabled={isFetchLoading}
                  label={inputLabel}
                  maxLength={BLOCK_CARD_SMS_CODE_LENGTH}
                  onChange={onChangeSecureCode}
                  value={secureCode}
                  autoFocus
                  inputMode="numeric"
                  isLoading={isFetchLoading}
                  type="text"
                  hasError={isSecureCodeInputError}
                />
              </ControlWithTips>
            </Box>
            {isTimerRunning && (
              <CodeRequestTimer
                isTimerRunning={isTimerRunning}
                setIsTimerRunning={setIsTimerRunning}
                text={timerInfo}
                timeout={timeout}
              />
            )}
          </Box>
        ))}
      <Box mb={INDENT.m}>
        {isSentCode ? (
          <ButtonContained.Loader
            data-test-id={getDataTestId(resendButton.type, 'blockCardButton')}
            disabled={isTimerRunning || isFetchLoading}
            isLoading={isFetchAthorizeLoading}
            onClick={handleResentSmsCode}
          >
            {resendButton.title}
          </ButtonContained.Loader>
        ) : (
          <ButtonContained.Loader
            data-test-id={getDataTestId(continueButton.type, 'blockCardButton')}
            onClick={handleSendSmsCode}
          >
            {continueButton.title}
          </ButtonContained.Loader>
        )}
      </Box>
      <Box>
        <BackButtonBlock
          data-test-id={getDataTestId(backButton.type, 'blockCardButton')}
          disabled={isFetchLoading || isFetchAthorizeLoading}
          onClick={onBack}
        >
          {backButton.title}
        </BackButtonBlock>
      </Box>
    </>
  );
};
