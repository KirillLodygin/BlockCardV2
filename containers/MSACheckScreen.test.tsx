import { shallow } from 'enzyme';
import React, { ChangeEvent, useState } from 'react';

import { ButtonContained } from '@vtb/ui-kit/adaptive/atoms/button';
import { Input } from '@vtb/ui-kit/adaptive/atoms/input';

import { BackButtonBlock } from '../components/components';
import { RowTableData, LOCK_OPERATION, BUTTONS_LOCK } from '../types';
import { MSACheckScreen } from './MSACheckScreen';

jest.mock('react', () => {
  const requireActual = jest.requireActual('react');

  return {
    ...requireActual,
    useState: jest.fn(),
  };
});

const mockOnResetSecureCodeError = jest.fn();
const mockSetIsSentCode = jest.fn();
const mockSetIsTimerRunning = jest.fn();
const mockOnBack = jest.fn();
const mockOnChangeSecureCode = jest.fn();
const mockOnSendSmsCode = jest.fn();

const mockEvent = {} as ChangeEvent<HTMLInputElement>;

const dataRowTableData: RowTableData[] = [
  {
    title: 'foo1',
    value: 'foo2',
  },
  {
    title: 'bar1',
    value: 'bar2',
  },
];

const defaultProps = {
  content: {
    backButton: {
      title: 'sdf',
      type: BUTTONS_LOCK.BACK,
    },
    continueButton: {
      title: 'xcv',
      type: BUTTONS_LOCK.CONTINUE,
    },
    resendButton: {
      title: 'cvb',
      type: BUTTONS_LOCK.RESEND,
    },
    dataRowTableData,
    inputErrorText: 'sts',
    inputLabel: 'asd',
    phoneInfo: 'bat',
    timerInfo: 'wer',
    title: 'qwe',
  },
  blockCardType: LOCK_OPERATION.PERMANENT,
  isFetchLoading: false,
  isFetchAthorizeLoading: false,
  isSecureCodeInputError: false,
  onBack: mockOnBack,
  onChangeSecureCode: mockOnChangeSecureCode,
  onResetSecureCodeError: mockOnResetSecureCodeError,
  onSendSmsCode: mockOnSendSmsCode,
  secureCode: 'zxc',
  timeout: 10,
};

describe('MSACheckScreen', () => {
  beforeEach(() => {
    (useState as jest.Mock).mockReset();
    mockOnBack.mockReset();
    mockOnChangeSecureCode.mockReset();
    mockOnSendSmsCode.mockReset();
    mockSetIsSentCode.mockReset();
    mockSetIsTimerRunning.mockReset();
    mockOnResetSecureCodeError.mockReset();
  });

  it('renders with all props', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [false, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders with isSentCode=true', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders with isSentCode=true and secureCode.length === 6', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} secureCode="qweasd" />);

    expect(component).toMatchSnapshot();
  });

  it('renders with isSentCode=true and isTimerRunning=false', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [false, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders with isSentCode=true and isFetchLoading=true', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} isFetchLoading />);

    expect(component).toMatchSnapshot();
  });

  it('calls onSentSmsCode', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [false, mockSetIsSentCode]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    component.find(ButtonContained.Loader).simulate('click');

    expect(mockOnSendSmsCode).toHaveBeenCalledTimes(1);
    expect(mockSetIsSentCode).toHaveBeenCalledTimes(1);
    expect(mockSetIsSentCode).toHaveBeenCalledWith(true);
  });

  it('calls onSendSmsCode with isSentCode=true and isTimerRunning=false', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [false, mockSetIsTimerRunning]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    component.find(ButtonContained.Loader).simulate('click');

    expect(mockOnSendSmsCode).toHaveBeenCalledTimes(1);
    expect(mockSetIsTimerRunning).toHaveBeenCalledTimes(1);
    expect(mockSetIsTimerRunning).toHaveBeenCalledWith(true);
  });

  it('calls onBack', () => {
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [false, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    component.find(BackButtonBlock).simulate('click');

    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });

  it('calls onChangeSecureCode', () => {
    const mockValue = 'quuux';
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);
    jest.spyOn(React, 'useState').mockImplementationOnce(() => [true, jest.fn()]);

    const component = shallow(<MSACheckScreen {...defaultProps} />);

    component.find(Input).prop('onChange')?.(mockValue, mockEvent);

    expect(mockOnChangeSecureCode).toHaveBeenCalledTimes(1);
    expect(mockOnChangeSecureCode).toHaveBeenCalledWith(mockValue, mockEvent);
  });
});
