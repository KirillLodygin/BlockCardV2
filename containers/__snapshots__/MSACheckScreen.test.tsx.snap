// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MSACheckScreen renders with all props 1`] = `
<Fragment>
  <BlockTitle
    titleText="qwe"
  />
  <styled.div
    mb="32px"
  >
    <DataRowTable
      data={
        Array [
          Object {
            "title": "foo1",
            "value": "foo2",
          },
          Object {
            "title": "bar1",
            "value": "bar2",
          },
        ]
      }
      size="m"
    />
  </styled.div>
  <styled.div
    mb="32px"
  >
    <ButtonContainedLoader
      data-test-id="continue_blockcardbutton"
      onClick={[Function]}
    >
      xcv
    </ButtonContainedLoader>
  </styled.div>
  <styled.div>
    <BackButtonBlock
      data-test-id="back_blockcardbutton"
      disabled={false}
      onClick={[MockFunction]}
    >
      sdf
    </BackButtonBlock>
  </styled.div>
</Fragment>
`;

exports[`MSACheckScreen renders with isSentCode=true 1`] = `
<Fragment>
  <BlockTitle
    titleText="qwe"
  />
  <styled.div
    mb="32px"
  >
    <DataRowTable
      data={
        Array [
          Object {
            "title": "foo1",
            "value": "foo2",
          },
          Object {
            "title": "bar1",
            "value": "bar2",
          },
        ]
      }
      size="m"
    />
  </styled.div>
  <styled.div
    mb="40px"
  >
    <styled.div
      mb="24px"
    >
      <AdditionalTypography
        size="m"
        tag="p"
        weight="normal"
      >
        bat
      </AdditionalTypography>
    </styled.div>
    <styled.div
      maxWidth="544px"
    >
      <ControlWithTips
        errorText="sts"
        hasError={false}
        width="352px"
      >
        <Input
          autoFocus={true}
          data-test-id="temporary_block_debit_card_input_smscode_field"
          disabled={false}
          hasError={false}
          inputMode="numeric"
          isLoading={false}
          label="asd"
          maxLength={6}
          onChange={[MockFunction]}
          type="text"
          value="zxc"
        />
      </ControlWithTips>
    </styled.div>
    <CodeRequestTimer
      isTimerRunning={true}
      setIsTimerRunning={[MockFunction]}
      text="wer"
      timeout={10}
    />
  </styled.div>
  <styled.div
    mb="32px"
  >
    <ButtonContainedLoader
      data-test-id="resend_blockcardbutton"
      disabled={true}
      isLoading={false}
      onClick={[Function]}
    >
      cvb
    </ButtonContainedLoader>
  </styled.div>
  <styled.div>
    <BackButtonBlock
      data-test-id="back_blockcardbutton"
      disabled={false}
      onClick={[MockFunction]}
    >
      sdf
    </BackButtonBlock>
  </styled.div>
</Fragment>
`;

exports[`MSACheckScreen renders with isSentCode=true and isFetchLoading=true 1`] = `
<Fragment>
  <BlockTitle
    titleText="qwe"
  />
  <styled.div
    mb="32px"
  >
    <DataRowTable
      data={
        Array [
          Object {
            "title": "foo1",
            "value": "foo2",
          },
          Object {
            "title": "bar1",
            "value": "bar2",
          },
        ]
      }
      size="m"
    />
  </styled.div>
  <styled.div
    mb="40px"
  >
    <styled.div
      mb="24px"
    >
      <AdditionalTypography
        size="m"
        tag="p"
        weight="normal"
      >
        bat
      </AdditionalTypography>
    </styled.div>
    <styled.div
      maxWidth="544px"
    >
      <ControlWithTips
        errorText="sts"
        hasError={false}
        width="352px"
      >
        <Input
          autoFocus={true}
          data-test-id="temporary_block_debit_card_input_smscode_field"
          disabled={true}
          hasError={false}
          inputMode="numeric"
          isLoading={true}
          label="asd"
          maxLength={6}
          onChange={[MockFunction]}
          type="text"
          value="zxc"
        />
      </ControlWithTips>
    </styled.div>
    <CodeRequestTimer
      isTimerRunning={true}
      setIsTimerRunning={[MockFunction]}
      text="wer"
      timeout={10}
    />
  </styled.div>
  <styled.div
    mb="32px"
  >
    <ButtonContainedLoader
      data-test-id="resend_blockcardbutton"
      disabled={true}
      isLoading={false}
      onClick={[Function]}
    >
      cvb
    </ButtonContainedLoader>
  </styled.div>
  <styled.div>
    <BackButtonBlock
      data-test-id="back_blockcardbutton"
      disabled={true}
      onClick={[MockFunction]}
    >
      sdf
    </BackButtonBlock>
  </styled.div>
</Fragment>
`;

exports[`MSACheckScreen renders with isSentCode=true and isTimerRunning=false 1`] = `
<Fragment>
  <BlockTitle
    titleText="qwe"
  />
  <styled.div
    mb="32px"
  >
    <DataRowTable
      data={
        Array [
          Object {
            "title": "foo1",
            "value": "foo2",
          },
          Object {
            "title": "bar1",
            "value": "bar2",
          },
        ]
      }
      size="m"
    />
  </styled.div>
  <styled.div
    mb="40px"
  >
    <styled.div
      mb="24px"
    >
      <AdditionalTypography
        size="m"
        tag="p"
        weight="normal"
      >
        bat
      </AdditionalTypography>
    </styled.div>
    <styled.div
      maxWidth="544px"
    >
      <ControlWithTips
        errorText="sts"
        hasError={false}
        width="352px"
      >
        <Input
          autoFocus={true}
          data-test-id="temporary_block_debit_card_input_smscode_field"
          disabled={false}
          hasError={false}
          inputMode="numeric"
          isLoading={false}
          label="asd"
          maxLength={6}
          onChange={[MockFunction]}
          type="text"
          value="zxc"
        />
      </ControlWithTips>
    </styled.div>
  </styled.div>
  <styled.div
    mb="32px"
  >
    <ButtonContainedLoader
      data-test-id="resend_blockcardbutton"
      disabled={false}
      isLoading={false}
      onClick={[Function]}
    >
      cvb
    </ButtonContainedLoader>
  </styled.div>
  <styled.div>
    <BackButtonBlock
      data-test-id="back_blockcardbutton"
      disabled={false}
      onClick={[MockFunction]}
    >
      sdf
    </BackButtonBlock>
  </styled.div>
</Fragment>
`;

exports[`MSACheckScreen renders with isSentCode=true and secureCode.length === 6 1`] = `
<Fragment>
  <BlockTitle
    titleText="qwe"
  />
  <styled.div
    mb="32px"
  >
    <DataRowTable
      data={
        Array [
          Object {
            "title": "foo1",
            "value": "foo2",
          },
          Object {
            "title": "bar1",
            "value": "bar2",
          },
        ]
      }
      size="m"
    />
  </styled.div>
  <styled.div
    mb="40px"
  >
    <styled.div
      mb="24px"
    >
      <AdditionalTypography
        size="m"
        tag="p"
        weight="normal"
      >
        bat
      </AdditionalTypography>
    </styled.div>
    <styled.div
      maxWidth="544px"
    >
      <ControlWithTips
        errorText="sts"
        hasError={false}
        width="352px"
      >
        <Input
          autoFocus={true}
          data-test-id="temporary_block_debit_card_input_smscode_field"
          disabled={false}
          hasError={false}
          inputMode="numeric"
          isLoading={false}
          label="asd"
          maxLength={6}
          onChange={[MockFunction]}
          type="text"
          value="qweasd"
        />
      </ControlWithTips>
    </styled.div>
    <CodeRequestTimer
      isTimerRunning={true}
      setIsTimerRunning={[MockFunction]}
      text="wer"
      timeout={10}
    />
  </styled.div>
  <styled.div
    mb="32px"
  >
    <ButtonContainedLoader
      data-test-id="resend_blockcardbutton"
      disabled={true}
      isLoading={false}
      onClick={[Function]}
    >
      cvb
    </ButtonContainedLoader>
  </styled.div>
  <styled.div>
    <BackButtonBlock
      data-test-id="back_blockcardbutton"
      disabled={false}
      onClick={[MockFunction]}
    >
      sdf
    </BackButtonBlock>
  </styled.div>
</Fragment>
`;
